common/prometheus-monitors/templates/podmonitor.yaml

```diff
@@ -0,0 +1,56 @@
+{{- if .Values.podMonitor.enabled }}
+{{- if .Values.serviceMonitor.enabled }}
+{{ fail "failed to render PodMonitor. ServiceMonitor is also enabed. Only one is allowed."}}
+{{- end }}
```

Comment:
```suggestion
{{- if and .Values.podMonitor.enabled .Values.serviceMonitor.enabled }}
{{ fail "Could not render Podmonitor. '.Values.podMonitor.enabled' and '.Values.serviceMonitor.enabled' are mutually exclusive."}}
{{- end }}
```

Reply:
This is intentionally split. The outer if is for determining if the template should be rendered the inner if to fail if both service and pod monitor are enabled.

Reply:
I see. Overlooked it at first glance.